src/api/__tests__/Metadata.test.js

```diff
@@ -3005,4 +3007,173 @@ describe('api/Metadata', () => {
             expect(handleOnAbort).toHaveBeenCalled();
         });
     });
+    describe('getMetadataTaxonomyLevels', () => {
+        const scope = 'enterprise';
+        const taxonomyKey = '12345';
+        const fileID = 'id';
+
+        test('should build getMetadataTaxonomyLevelsUrl correctly', () => {
+            const url = metadata.getMetadataTaxonomyLevelsUrl(scope, taxonomyKey);
+
+            expect(url).toBe(`https://api.box.com/2.0/metadata_taxonomies/${scope}/${taxonomyKey}`);
+        });
+
+        test('should fetch metadata taxonomy levels successfully', async () => {
+            const mockResponse = {
+                displayName: 'Geography',
+                namespace: 'my_enterprise',
+                id: 'this-is-a-taxonomy-id',
+                key: 'geography',
+                levels: [
+                    {
+                        displayName: 'Independent Nations States',
+                        description: 'Country',
+                        level: 1,
+                    },
+                    {
+                        displayName: 'States of a Specific Country',
+                        description: 'State',
+                        level: 2,
+                    },
+                ],
+            };
+            metadata.xhr.get = jest.fn().mockReturnValueOnce({ data: mockResponse });
+            metadata.getMetadataTaxonomyLevelsUrl = jest.fn().mockReturnValueOnce('levels_url');
+
+            const result = await metadata.getMetadataTaxonomyLevels(fileID, scope, taxonomyKey);
+
+            expect(result).toEqual(mockResponse);
+            expect(metadata.errorCode).toBe(ERROR_CODE_FETCH_METADATA_TAXONOMY_LEVELS);
+            expect(metadata.getMetadataTaxonomyLevelsUrl).toHaveBeenCalled();
+            expect(metadata.xhr.get).toHaveBeenCalledWith({
+                id: 'file_id',
+                url: 'levels_url',
+            });
+        });
+
+        test('should throw an error if id is missing', async () => {
```

Comment:
nit: We could put the following three tests into a single test using `test.each`

Reply:
Not sure if the value is worth it since both the arguments and results are slightly different.